### PR TITLE
Add InterQuartileTransformer normalization strategy for Number Features

### PR DESCRIPTION
**Pull request by arnavgarg1**
This PR adds a new normalization method for numeric features called `InterQuartileTransformer`, which removes the median and scales the data according to the interquartile range. The resulting data has a zero mean and median and a standard deviation of 1, and it is not skewed by outliers (the outliers are still present with the same relative relationships to other values).

This effectively implements https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.RobustScaler.html, which is useful in cases where numeric features have large outliers that skew the distribution that emerges post min-max normalization.

This can be set in Ludwig by specifying `iq` as the normalization method in number feature preprocessing

### CODE REVIEW

File: ludwig/features/number_feature.py [hunk 1]
```diff
@@ -102,6 +102,45 @@ def fit_transform_params(column: np.ndarray, backend: "Backend") -> dict:  # noq
...
+    def inverse_transform_inference(self, x: torch.Tensor) -> torch.Tensor:
+        return x * self.interquartile_range + self.q2
+
+    @staticmethod
+    def fit_transform_params(column: np.ndarray, backend: "Backend") -> dict:  # noqa
+        compute = backend.df_engine.compute
+        return {
+            "q1": compute(np.percentile(column.astype(np.float32), 25)),
+            "q2": compute(np.percentile(column.astype(np.float32), 50)),
+            "q3": compute(np.percentile(column.astype(np.float32), 75)),
```
Comment (justinxzhao) [inline ludwig/features/number_feature.py:140]: Just an idea for a future improvement -- we may be able to get this information from the dataset profile directly instead of having to calculate it anew during preprocessing.

Comment (arnavgarg1) [reply ludwig/features/number_feature.py:140]: We may want to calculate all of these statistics in advance then (even for the other transformers (e.g., mean, stddev, min, max, log1p etc)

**Comment by arnavgarg1** [comment]
Tested this with Titanic with @justinxzhao. I commented out the `parch` feature since it has an interquartile range of 0 which would cause a DivisionByZero error. Results look reasonable, will merge.
